Lib/dataclasses.py

```diff
@@ -0,0 +1,776 @@
+    # Pre-compute self_tuple and other_tuple, then re-use them for
+    #  each function.
+    self_tuple = _tuple_str('self', fields)
+    other_tuple = _tuple_str('other', fields)
+    for name, op in [('__eq__', '=='),
+                     ('__ne__', '!='),
+                     ]:
+        _set_attribute(cls, name, _cmp_fn(name, op, self_tuple, other_tuple))
+
+
+def _set_order_fns(cls, fields):
+    # Create and set the ordering methods on cls.
+    # Pre-compute self_tuple and other_tuple, then re-use them for
+    #  each function.
+    self_tuple = _tuple_str('self', fields)
+    other_tuple = _tuple_str('other', fields)
+    for name, op in [('__lt__', '<'),
+                     ('__le__', '<='),
+                     ('__gt__', '>'),
+                     ('__ge__', '>='),
+                     ]:
+        _set_attribute(cls, name, _cmp_fn(name, op, self_tuple, other_tuple))
+
+
+def _hash_fn(fields):
+    self_tuple = _tuple_str('self', fields)
+    return _create_fn('__hash__',
+                      ['self'],
+                      [f'return hash({self_tuple})'])
+
+
+def _get_field(cls, a_name, a_type):
+    # Return a Field object, for this field name and type.  ClassVars
+    #  and InitVars are also returned, but marked as such (see
+    #  f._field_type).
+
+    # If the default value isn't derived from field, then it's
+    #  only a normal default value.  Convert it to a Field().
+    default = getattr(cls, a_name, _MISSING)
+    if isinstance(default, Field):
+        f = default
+    else:
+        f = field(default=default)
+
+    # Assume it's a normal field until proven otherwise.
+    f._field_type = _FIELD
+
+    # Only at this point do we know the name and the type. Set them.
+    f.name = a_name
+    f.type = a_type
+
+    # If typing has not been imported, then it's impossible for
+    #  any annotation to be a ClassVar. So, only look for ClassVar
+    #  if typing has been imported.
+    typing = sys.modules.get('typing')
+    if typing is not None:
+        # This test uses a typing internal class, but it's the best
+        #  way to test if this is a ClassVar.
+        if type(a_type) is typing._ClassVar:
+            # This field is a ClassVar, so it's not a field.
+            f._field_type = _FIELD_CLASSVAR
+
+    if f._field_type is _FIELD:
+        # Check if this is an InitVar.
+        if a_type is InitVar:
+            # InitVars are not fields, either.
+            f._field_type = _FIELD_INITVAR
+
+    # Validations for fields.  This is delayed until now, instead of
+    # in the Field() constructor, since only here do we know the field
+    # name, which allows better error reporting.
+
+    # Special restrictions for ClassVar and InitVar.
+    if f._field_type in (_FIELD_CLASSVAR, _FIELD_INITVAR):
+        if f.default_factory is not _MISSING:
+            raise TypeError(f'field {f.name} cannot have a '
+                            'default factory')
```

Comment:
f?

Reply:
I'm not a big fan of the extra f's. It requires the compiler to do a little more work (admittedly, it's probably nothing in the scheme of things).